### DISCUSSION TITLE
The Array Returned by The Child Component won't Be Rendered

### DISCUSSION THREAD

**Discussion by cqh963852** [vanjs-org/van#117] [Q&A] [answered]
Greatings, I am writing some app with van jsx.

I am facing a problem. 

While a function child returns an array. It won't be rendered.

```
const Todo = () => {
  const items = van.state<number[]>([]);
  let count = 0;
  return (
    <div>
      <button
        onclick={() => {
          items.val = [...items.oldVal, count++];
        }}
      >
        add item
      </button>

      <h1>1.1</h1>
      {/* 1.1 */}
      {() => {
        return van.tags.div(
          items.val.map((value) => {
            return van.tags.div({
              textContent: value,
            });
          })
        );
      }}

      <h1>1.2</h1>
      {/* 1.2 */}
      {() => {
        return (
          <div>
            {items.val.map((value) => {
              return <div>{value}</div>;
            })}
          </div>
        );
      }}

      <h1>2.1</h1>
      {/* 2.1 */}
      {() => {
        return items.val.map((value) => {
          return van.tags.div({
            value,
          });
        });
      }}

      <h1>2.2</h1>
      {/* 1.4 */}
      {() => {
        return items.val.map((value) => {
          return <div>{value}</div>;
        });
      }}
    </div>
  );
};
```
![image](https://github.com/vanjs-org/van/assets/17702287/5552534c-22d8-48a2-9669-3e78234bda29)

---

It seems vanjs-core treat the return of the function child as `a dom`. Not `dom[] or dom`

![image](https://github.com/vanjs-org/van/assets/17702287/46cab8cb-1d43-4bd5-8610-6e69c08cfd3c)

---

I would like to ask if it would be a good idea to add array-related rendering implementation here ?

**Comment by cqh963852** [comment]
If want to support functions that return array-type components.

It will need to do a lot of array comparison work. 

This may not be conducive to keeping vanjs small in size.

**Comment by Tao-VanJS** [comment, marked as answer]
You can wrap the result into a pass-through container (`span` for inline elements and `div` for block elements) if multiple elements need to be returned.

See https://vanjs.org/demo#blog.
